lemminx-maven/src/main/java/org/eclipse/lemminx/extensions/maven/MavenLemminxExtension.java

```diff
@@ -328,9 +330,16 @@ public MavenPluginManager getMavenPluginManager() {
 		initialize();
 		return mavenPluginManager;
 	}
-
+	
 	public Optional<RemoteRepositoryIndexSearcher> getIndexSearcher() {
 		initialize();
 		return Optional.ofNullable(indexSearcher);
 	}
+
+	public void didChangeWorkspaceFolders(URI[] added, URI[] removed) {
+		WorkspaceReader workspaceReader = mavenRequest.getWorkspaceReader();
+		if (workspaceReader instanceof MavenLemminxWorkspaceReader) {
+			// TODO

```

Comment:
This should actually process all the workspaceFolders pom.xml files to add them in the cache.

Reply:
Does it mean that I have to recursively parse at least all the XML files that exist in the added folders in order to find and add pom files to the Project Cache?

Reply:
That would actually be nice. However, I think it's not necessary to implement this "TODO" for this commit, let's ignore it at the moment and we'll implement it in further commits.

Reply:
ok